src/lib/plugin.js

```diff
@@ -157,21 +157,49 @@ module.exports = class PluginVirtual extends EventEmitter2 {
   }
 
   async sendRequest (message) {
+    this._validator.validateOutgoingMessage(message)
+    this._safeEmit('outgoing_request', message)
+
     // for backward compat with ilp-kit version < 2.1.0, copy `.custom` to `.data`.
     // This will only work in connector-to-connector send_message rpc calls as implemented here in
     // ilp-plugin-virtual. When sent over a FiveBellsLedger, like in five-bells-integration-test,
     // message.data will be stripped again by ilp-plugin-bells, because it does not comply with
     // the Message schema as defined by five-bells-shared.
-    message.data = message.custom
-
-    this._validator.validateOutgoingMessage(message)
-    this._safeEmit('outgoing_request', message)
+    // Using JSON stringify-then-parse here as an object cloning method, since Object.assign only
+    // makes a shallow copy, and message.custom is expected to have nested objects in it.
+    const convertedMessage = JSON.parse(JSON.stringify(message))
+    convertedMessage.data = convertedMessage.custom
+    delete convertedMessage.custom
+    const response = await this._rpc.call('send_message', this._prefix, [convertedMessage])
+    // convert a response from send_message endpoint to response format of send_request endpoint
+    const convertedResponse = {
+      ledger: message.ledger,
+      from: message.to,
+      to: message.from
+    }
+    if (response !== true) {
+      convertedResponse.ilp = IlpPacket.serializeIlpError({
+        triggeredBy: message.to,
+        forwardedBy: [ message.from ],
+        code: 'T00',
+	name: response
+      })
+    }
+    this._safeEmit('incoming_response', convertedResponse)
+    return convertedResponse
+  }
 
-    const response = await this._rpc.call('send_message', this._prefix, [message])
-    this._validator.validateIncomingMessage(response)
-    this._safeEmit('incoming_response', response)
+  async _handleMessage (message) {
+    // convert messages from ilp-kit version < 2.1.0
+    const convertedMessage = JSON.parse(JSON.stringify(message))
+    convertedMessage.custom = convertedMessage.data
```

Comment:
Instead of sending just the old `data` field over RPC, we should send both `data` and `custom`, and then eventually stop transmitting `data`